core/src/main/java/org/radargun/stages/cache/test/GaussianKeysSelector.java

```diff
@@ -0,0 +1,63 @@
+package org.radargun.stages.cache.test;
+
+import java.util.Random;
+
+import org.radargun.config.DefinitionElement;
+import org.radargun.config.Property;
+import org.radargun.config.PropertyHelper;
+
+/**
+ * @author Jakub Markos &lt;jmarkos@redhat.com&gt;
+ */
+public class GaussianKeysSelector implements KeySelector {
+    private final Random random;
+    private final long numEntries;
+    private final long mean;
+    private final long standardDeviation;
+
+    public GaussianKeysSelector(Random random, long numEntries, long mean, long standardDeviation) {
+        this.random = random;
+        this.numEntries = numEntries;
+        this.mean = mean;
+        this.standardDeviation = standardDeviation;
+    }
+
+    @Override
+    public long next() {
+        long result = (long) (random.nextGaussian() * standardDeviation + mean);
+        while (result < 0 || result >= numEntries) {
+            result = (long) (random.nextGaussian() * standardDeviation + mean);
+        }
+        return result;
+    }
+
+    @DefinitionElement(name = "gaussian-keys", doc = "In the test fixed set of entries is used and this is shared among all stressor threads. Additionally, the keys" +
+            "are selected using a normal (gaussian) distribution.")
+    public static class Factory implements KeySelectorFactory {
+        @Property(doc = "Total number of key-value entries.", optional = false)
+        private long numEntries = 0;
+
+        @Property(doc = "Mean value of indices. Default is numEntries/2.")
+        private long mean = -1;
+
+        @Property(doc = "Standard deviation. Default is numEntries/8.")
+        private long standardDeviation = -1;
+
+        @Override
+        public KeySelector newInstance(CacheOperationsTestStage stage, Random random, int globalThreadId, int threadId) {
+            if (mean == -1) {
```

Comment:
I think you can initialize the instance variables with their values directly in declaration. If you specify property value, it has a precedence over default value.